benches/comparative.rs

```diff
@@ -0,0 +1,66 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::prelude::*;
+use std::collections::HashSet;
+
+const SIZE: usize = 10000;
+
+/// This benchmark compares the performance of HashSet, Vob, and Vec
+/// in a scenario where an algorithm needs to keep track of visited indices.
+#[cfg(test)]
+fn comparative(c: &mut Criterion) {
+    // Generate 10000 random (seeded) points
+    let mut rng: StdRng = SeedableRng::from_seed([42; 32]);
+    let mut indices = Vec::<usize>::new();
```

Comment:
I think this can be something like `let mut indices = 0..SIZE..collect()`?

Reply:
Good catch, fixed. 
I'd like to rebase the PR before release though